build-data-capturing-gradle-samples/capture-test-pts-support/gradle-test-pts-support.gradle

```diff
@@ -58,6 +59,19 @@ class Capture {
         }
     }
 
+    boolean ptsSupported(Test t, Set<String> engines) {
+        if (!engines.isEmpty() && engines.stream().allMatch { e -> supportedEngines.containsKey(e) }) {
+            // If cucumber is used without companion it's not supported, otherwise it is.
+            return !(cucumberUsed(t, engines) && !t.project.plugins.hasPlugin('com.gradle.cucumber.companion'))
```

Comment:
Nope, this is the function that is checking if PTS is supported. So, if cucumber is used and no companion plugin is applied, PTS is NOT supported. If there's no cucumber found or if cucumber is used with companion, the PTS is supported.

Reply:
Thanks for explaining. I'm following the logic now. I think the `if` and nested boolean conditionals makes it a little confusing to read. It would be nice if this could be updated to all be at the same level of abstraction.

Reply:
Hopefully I improved readability.. :)

Reply:
Better? :)